src/widgets/modals/ByProgramModal.js

```diff
@@ -1,376 +1,248 @@
+        )}
+      </PageContentModal>
     );
   };
 
-  renderModal = () => {
-    const { currentStepKey } = this.state;
-    if (currentStepKey === 'name') return this.renderTextEditor();
-    return <AutocompleteSelector {...this.getSearchModalProps()[currentStepKey]} />;
+  const stepProps = {
+    program: useMemo(
+      () => ({
+        data: program,
+        getModalData: getPrograms,
+        onPress: onOpenModal,
+        status: getStatus('program'),
+        stepKey: 'program',
+        type: 'select',
+        field: 'name',
+      }),
+      [program, state.isProgramBased]
+    ),
+    supplier: useMemo(
+      () => ({
+        data: supplier,
+        getModalData: getSuppliers,
+        onPress: onOpenModal,
+        status: getStatus('supplier'),
+        stepKey: 'supplier',
+        type: 'select',
+        field: 'name',
+      }),
+      [program, supplier, state.isProgramBased]
+    ),
+    orderType: useMemo(
+      () => ({
+        data: orderType,
+        getModalData: getOrderTypes,
+        onPress: onOpenModal,
+        status: getStatus('orderType'),
+        stepKey: 'orderType',
+        type: 'select',
+        field: 'name',
+      }),
+      [supplier, orderType]
+    ),
+    period: useMemo(
+      () => ({
+        data: period,
+        getModalData: getPeriods,
+        onPress: onOpenModal,
+        status: getStatus('period'),
+        stepKey: 'period',
+        type: 'select',
+        field: 'name',
+      }),
+      [orderType, period]
+    ),
+    name: useMemo(
+      () => ({
+        data: name,
+        status: getStatus('name'),
+        onPress: onOpenModal,
+        stepKey: 'name',
+        type: 'input',
+      }),
+      [program, state.isProgramBased, name]
+    ),
   };
 
-  renderToggleBar = () => {
-    const { type } = this.props;
-    const { isProgramBased } = this.state;
-    const buttonText =
-      type === 'requisition' ? this.LOCALIZATION.button.order : this.LOCALIZATION.button.stocktake;
-    return (
-      <ToggleBar
-        toggles={[
-          {
-            text: `${this.LOCALIZATION.misc.program} ${buttonText}`,
-            onPress: this.onToggleChange,
-            isOn: isProgramBased,
-          },
-          {
-            text: `${this.LOCALIZATION.misc.general} ${buttonText}`,
-            onPress: this.onToggleChange,
-            isOn: !isProgramBased,
-          },
-        ]}
+  /** Render */
+  const { modalStyle, okButton, pageButtonTextStyle } = localStyles;
+  const { isModalOpen } = state;
+  const { isOpen, onCancel } = props;
+  const { currentKey } = state;
+  return (
+    <PageContentModal isOpen={isOpen} style={modalStyle} swipeToClose={false} onClose={onCancel}>
+      <ProgramToggleBar />
+
+      {steps.map(stepKey => (
+        <Step key={stepKey} {...stepProps[stepKey]} />
+      ))}
+
+      <PageButton
+        text="OK"
+        onPress={onCreate}
+        isDisabled={!(steps[steps.length - 1] === currentKey)}
```

Comment:
Might be tidier to put this into its own variable:

```
const isDisabled = {!(steps[steps.length - 1] === currentKey)}
...
return (
  ...
 <PageButton
        text="OK"
        onPress={onCreate}
        isDisabled={isDisabled}
        ...
  ...
)
```
```

Reply:
0b18c0556fad9ca50e797b4cd7279faf710e596f